logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueWriterTest.java

```diff
@@ -92,4 +98,33 @@ public void testWrite() throws Exception {
         writer.writeEntry(entry);
         writer.close();
     }
+
+    @Test
+    public void testDoesNotWriteBeyondLimit() throws Exception {
+        DLQEntry entry = new DLQEntry(new Event(), "type", "id", "reason");
+
+        int payloadLength = RECORD_HEADER_SIZE + VERSION_SIZE + entry.serialize().length;
+        long queueLength = payloadLength * 5;
```

Comment:
@robbavey I think it would be nice for readabilities sake if we extract the `5` int a variable right?